src/hookEvent.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef DISPLAZ_HOOKEVENT_H_INCLUDED
+#define DISPLAZ_HOOKEVENT_H_INCLUDED
+
+#include <QByteArray>
+#include <QKeySequence>
+#include <QShortcut>
+
+/// Class for registering custom hook events. Implemented via QShortCut,
+/// i.e. for QKeySequences only.
+class hookEvent : public QObject
+{
+    Q_OBJECT
+    public:
+        hookEvent(QObject* parent = NULL);
+
+        void setHookEvent(QByteArray hookSpec);
+        int getHookId() const { return latestHookIndex; }
+        QVector<QShortcut*> m_shortCut;
+
```

Comment:
Having thought about it, I think you can put these three public things together into one.  Something roughly like

```C++
/// Connect events defined by `eventSpec` to the HookFormatter hookActivated() function.
void connectHook(QByteArray eventSpec, HookFormatter* formatter)
```